kyo-prelude/jvm/src/test/scala/kyo2/kernel/BytecodeTest.scala

```diff
@@ -0,0 +1,55 @@
+package kyo2.kernel
+
+import kyo.Tag
+import kyo2.*
+import scala.reflect.ClassTag
+
+class BytecodeTest extends Test:
+
+    object TestEffect extends Effect[Id, Id]
+
+    class TestSuspend:
+        def test() = Effect.suspend[Int](Tag[TestEffect.type], 42)
+
+    class TestMap:
+        def test(v: Int < TestEffect.type) = v.map(_ + 1)
+
+    class TestHandle:
+        def test(v: Int < TestEffect.type) = Effect.handle(Tag[TestEffect.type], v)([C] => (input, cont) => cont(input))
+
+    "suspend" in {
```

Comment:
The new kernel generates larger trees to keep the trace information and provide stack safety. I'll follow up later to try reduce the generated bytecode size.